src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/network/v2/AwsV2NetworkPluginTest.java

```diff
@@ -0,0 +1,616 @@
+
+		Mockito.when(this.sharedOrderHolders.getActiveOrdersMap()).thenReturn(new HashMap<>());
+	}
+	
+	// test case: When calling the requestInstance method, with a network order and
+	// cloud user valid, a client is invoked to create a network instance, returning
+	// the network ID.
+	@Test
+	public void testRequestInstanceSuccessful() throws FogbowException {
+		// set up
+		Ec2Client client = Mockito.mock(Ec2Client.class);
+		PowerMockito.mockStatic(AwsV2ClientUtil.class);
+		BDDMockito.given(AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+
+		String vpcId = FAKE_VPC_ID;
+		mockCreateSubnets(null, client);
+		mockCreateRouteTables(vpcId, client);
+		mockCreateSecurityGroups(client);
+
+		NetworkOrder networkOrder = Mockito.spy(new NetworkOrder());
+		AwsV2User cloudUser = Mockito.mock(AwsV2User.class);
+
+		String expected = FAKE_SUBNET_ID;
+
+		// exercise
+		String networkId = this.plugin.requestInstance(networkOrder, cloudUser);
+
+		// verify
+		PowerMockito.verifyStatic(AwsV2ClientUtil.class, VerificationModeFactory.times(1));
+		AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString());
+
+		Mockito.verify(client, Mockito.times(1)).createSubnet(Mockito.any(CreateSubnetRequest.class));
+		Mockito.verify(client, Mockito.times(1)).associateRouteTable(Mockito.any(AssociateRouteTableRequest.class));
+		Mockito.verify(client, Mockito.times(1)).createSecurityGroup(Mockito.any(CreateSecurityGroupRequest.class));
+		Mockito.verify(client, Mockito.times(2)).createTags(Mockito.any(CreateTagsRequest.class));
+		Mockito.verify(client, Mockito.times(1))
+				.authorizeSecurityGroupIngress(Mockito.any(AuthorizeSecurityGroupIngressRequest.class));
+
+		Assert.assertEquals(expected, networkId);
+	}
+	
+	// test case: When calling the deleteInstance method, with a network order and
+	// cloud user valid, the network instance in the cloud must be deleted.
+	@Test
+	public void testDeleteInstanceSuccessful() throws FogbowException {
+		// set up
+		Ec2Client client = Mockito.mock(Ec2Client.class);
+		PowerMockito.mockStatic(AwsV2ClientUtil.class);
+		BDDMockito.given(AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+
+		mockDescribeSubnets(client);
+
+		NetworkOrder networkOrder = Mockito.spy(new NetworkOrder());
+		AwsV2User cloudUser = Mockito.mock(AwsV2User.class);
+
+		// exercise
+		this.plugin.deleteInstance(networkOrder, cloudUser);
+
+		// verify
+		PowerMockito.verifyStatic(AwsV2ClientUtil.class, VerificationModeFactory.times(1));
+		AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString());
+		
+		Mockito.verify(client, Mockito.times(1)).describeSubnets(Mockito.any(DescribeSubnetsRequest.class));
+		Mockito.verify(client, Mockito.times(1)).deleteSecurityGroup(Mockito.any(DeleteSecurityGroupRequest.class));
+		Mockito.verify(client, Mockito.times(1)).deleteSubnet(Mockito.any(DeleteSubnetRequest.class));
```

Comment:
I think this tests could be improved as follows: .deleteSubnet(deleteSubRequest). Where deleteSubRequest would be created before, passing the subnetId to its builder. Thus, we guarantee that the right parameter was passed to deleteSubnet.

Reply:
It can be done to the other tests too, check if it makes sense, if so, update all tests that it can be done please.